From 1253edcbf892b84e063c28ace3e319e37be452dc Mon Sep 17 00:00:00 2001
From: Eric Courtois <eric.courtois@gmail.com>
Date: Sun, 10 Jul 2022 19:06:15 +0200
Subject: [PATCH] Add generated Protobuf extension file

Using EmbeddedProto requires to run setup.sh that generate the Protobuf
extension file.
Since we do not run setup.sh, add this file in the sources.

Signed-off-by: Eric Courtois <eric.courtois@gmail.com>

diff --git a/.gitignore b/.gitignore
index f61e00e..fede341 100644
--- a/.gitignore
+++ b/.gitignore
@@ -42,4 +42,4 @@ build-wrapper**
 code_coverage_report/**

 # Ignore the generated options file.
-generator/embedded_proto_options_pb2.py
+#generator/embedded_proto_options_pb2.py
diff --git a/generator/embedded_proto_options_pb2.py b/generator/embedded_proto_options_pb2.py
new file mode 100644
index 0000000..285c508
--- /dev/null
+++ b/generator/embedded_proto_options_pb2.py
@@ -0,0 +1,82 @@
+# Generated by the protocol buffer compiler.  DO NOT EDIT!
+# source: embedded_proto_options.proto
+
+import sys
+_b=sys.version_info[0]<3 and (lambda x:x) or (lambda x:x.encode('latin1'))
+from google.protobuf import descriptor as _descriptor
+from google.protobuf import message as _message
+from google.protobuf import reflection as _reflection
+from google.protobuf import symbol_database as _symbol_database
+# @@protoc_insertion_point(imports)
+
+_sym_db = _symbol_database.Default()
+
+
+from google.protobuf import descriptor_pb2 as google_dot_protobuf_dot_descriptor__pb2
+
+
+DESCRIPTOR = _descriptor.FileDescriptor(
+  name='embedded_proto_options.proto',
+  package='EmbeddedProto',
+  syntax='proto3',
+  serialized_options=None,
+  serialized_pb=_b('\n\x1c\x65mbedded_proto_options.proto\x12\rEmbeddedProto\x1a google/protobuf/descriptor.proto\"\x1c\n\x07Options\x12\x11\n\tmaxLength\x18\x01 \x01(\r:G\n\x07options\x12\x1d.google.protobuf.FieldOptions\x18\xf5\x08 \x01(\x0b\x32\x16.EmbeddedProto.Optionsb\x06proto3')
+  ,
+  dependencies=[google_dot_protobuf_dot_descriptor__pb2.DESCRIPTOR,])
+
+
+OPTIONS_FIELD_NUMBER = 1141
+options = _descriptor.FieldDescriptor(
+  name='options', full_name='EmbeddedProto.options', index=0,
+  number=1141, type=11, cpp_type=10, label=1,
+  has_default_value=False, default_value=None,
+  message_type=None, enum_type=None, containing_type=None,
+  is_extension=True, extension_scope=None,
+  serialized_options=None, file=DESCRIPTOR)
+
+
+_OPTIONS = _descriptor.Descriptor(
+  name='Options',
+  full_name='EmbeddedProto.Options',
+  filename=None,
+  file=DESCRIPTOR,
+  containing_type=None,
+  fields=[
+    _descriptor.FieldDescriptor(
+      name='maxLength', full_name='EmbeddedProto.Options.maxLength', index=0,
+      number=1, type=13, cpp_type=3, label=1,
+      has_default_value=False, default_value=0,
+      message_type=None, enum_type=None, containing_type=None,
+      is_extension=False, extension_scope=None,
+      serialized_options=None, file=DESCRIPTOR),
+  ],
+  extensions=[
+  ],
+  nested_types=[],
+  enum_types=[
+  ],
+  serialized_options=None,
+  is_extendable=False,
+  syntax='proto3',
+  extension_ranges=[],
+  oneofs=[
+  ],
+  serialized_start=81,
+  serialized_end=109,
+)
+
+DESCRIPTOR.message_types_by_name['Options'] = _OPTIONS
+DESCRIPTOR.extensions_by_name['options'] = options
+_sym_db.RegisterFileDescriptor(DESCRIPTOR)
+
+Options = _reflection.GeneratedProtocolMessageType('Options', (_message.Message,), dict(
+  DESCRIPTOR = _OPTIONS,
+  __module__ = 'embedded_proto_options_pb2'
+  # @@protoc_insertion_point(class_scope:EmbeddedProto.Options)
+  ))
+_sym_db.RegisterMessage(Options)
+
+options.message_type = _OPTIONS
+google_dot_protobuf_dot_descriptor__pb2.FieldOptions.RegisterExtension(options)
+
+# @@protoc_insertion_point(module_scope)
